### PR TITLE
Added Amazon Linux Support. Bumped chef and supported OS versions

### PR DESCRIPTION
**Pull request by rwinfosec**
*Bumped to Chef 14.
*Added Support for Amazon Linux 2018.03 and 2
*Updated and fixed support for Ubuntu 16.04 and 18.04
*Updated support for OSX to 10.13

### CODE REVIEW

File: recipes/amazon.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: osquery
+# Recipe:: amazon
+#
+# Copyright 2019, Dwolla Inc.
```
Comment (jacknagz) [inline recipes/amazon.rb]: Can you remove this copyright? It goes against the Apache 2 License

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Copyright 2013-2014 Jack Naglieri <jacknagzdev@gmail.com>
```
Comment (jacknagz) [inline LICENSE]: I'll add the LICENSE in a follow-up PR, along with CONTRIBUTING, and more.

File: recipes/amazon.rb (hunk 1 above)
Comment (rwinfosec) [reply recipes/amazon.rb]: Yep! No problem.

File: LICENSE (hunk 2 above)
Comment (rwinfosec) [reply LICENSE]: With the LICENSE file, it is required by FoodCritic and will fail the TravisCI build. I will just add an empty LICENSE file.

**Comment by rwinfosec** [comment]
Regards to testing, I've updated some of the unit tests to reflect the versions that I am testing.
I've verified all integration tests are passing with each vagrant image that is supported and added an additional integration test to verify that the osquery configuration file JSON content is as expected. I've included the changes to this cookbook into another and converged with an AWS Linux 2018.03 instance in AWS to verify that it is working as intended.

**Pull request merged**